### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by pdevine**
This change adds a makefile and a dockerfile so you can contain the parrot.

### CODE REVIEW

**Comment by jduss4** [comment]
Nothing can contain that much partying!!!

**Comment by pdevine** [comment]
@jduss4 true, but if there's one thing that's better than a partying parrot, it's a partying parrot on a whale.

See for yourself!  `docker run -it --rm pdevine/partyparrot`

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM scratch
```
Comment (robbyoconnor) [inline Dockerfile]: Use a [multistage build](https://docs.docker.com/develop/develop-images/multistage-build/) and build this inside of Docker entirely.

Comment (jmhobbs) [reply Dockerfile]: Something like this should work,
```dockerfile
FROM golang AS gobuilder
RUN go get -u github.com/jmhobbs/terminal-parrot

FROM scratch
COPY --from=gobuilder /go/bin/terminal-parrot /terminal-parrot
CMD ["/terminal-parrot"]
```

Though that totally ignores the working dir and gets it from the repo on build.  Would also want a dockerignore in that case.

**Comment by jmhobbs** [comment]
Sorry I let this one languish.  I'd love to get it tidy'd up and merged in!

File: Dockerfile (hunk 1 above)
Comment (robbyoconnor) [reply Dockerfile]: I wouldn't do `go get` -- I'd just use the current working directory and build.

Comment (pdevine) [reply Dockerfile]: I'm chuckling to myself because multistage builds didn't exist when I first submitted the PR.  :-D

Comment (robbyoconnor) [reply Dockerfile]: See the docs I linked -- use the working directory.

Comment (robbyoconnor) [reply Dockerfile]: @pdevine well they do now and it's not that difficult to handle :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM golang:alpine3.7
```
Comment (robbyoconnor) [inline Dockerfile]: FROM golang as builder

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+WORKDIR /project
```
Comment (robbyoconnor) [inline Dockerfile]: er...`GOPATH`

File: Dockerfile (hunk 1 above)
Comment (pdevine) [reply Dockerfile]: @robbyoconnor no argument there.

File: Dockerfile (hunk 2 above)
Comment (robbyoconnor) [reply Dockerfile]: Note: it doesn't matter what image you use as the first stage as it gets blown away later.

Comment (pdevine) [reply Dockerfile]: @robbyoconnor I'm very familiar with multistage builds, thanks. :-D

Comment (pdevine) [reply Dockerfile]: There's no reason to pull in all of the golang image in to build this.  The alpine image is less than half the size.  Even it should go on a diet.

File: Dockerfile (hunk 3 above)
Comment (pdevine) [reply Dockerfile]: Not sure what the concern is here.

**Comment by jmhobbs** [comment]
It builds, it runs, I'm merging.  We can argue the finer points in another PR if we really want to.

Thanks!

**Comment by robbyoconnor** [comment]
Would you be opposed to me making the changes @jmhobbs ?

**Comment by jmhobbs** [comment]
@robbyoconnor PR's accepted!  I'm not a docker expert so get in the weeds on what you're changing 😄

**Comment by robbyoconnor** [comment]
I happen to be pretty decent with Docker :)

**Comment by pdevine** [comment]
I just work there, so.. ¯\_(ツ)_/¯

**Pull request merged**